antarest/tools/lib.py

```diff
@@ -233,12 +241,41 @@ def extract_commands(study_path: Path, commands_output_dir: Path) -> None:
         )
 
 
-def generate_diff(base: Path, variant: Path, output_dir: Path) -> None:
+def generate_diff(
+    base: Path, variant: Path, output_dir: Path, version: Optional[str] = None
+) -> None:
     if not output_dir.exists():
         output_dir.mkdir(parents=True)
     matrices_dir = output_dir / MATRIX_STORE_DIR
     matrices_dir.mkdir(exist_ok=True)
 
+    with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
you can create a new directory inside the output_dir ("empty_base")

Reply:
also if using the tempfile.TemporaryDirectory (using the output_dir as prefix), it should wrap the whole content of the method, if not it will be deleted before eventual usage.